python/dgl/sparse/elementwise_op_diag.py

```diff
@@ -157,14 +158,50 @@ def diag_div(D1: DiagMatrix, D2: Union[DiagMatrix, Scalar]) -> DiagMatrix:
         )
         return diag(D1.val / D2.val, D1.shape)
     elif is_scalar(D2):
-        assert D2 != 0, "Division by zero is not allowed."
         return diag(D1.val / D2, D1.shape)
     else:
         # Python falls back to D2.__rtruediv__(D1) then TypeError when
         # NotImplemented is returned.
         return NotImplemented
 
 
+def diag_rdiv(D1: DiagMatrix, D2: Union[DiagMatrix, Scalar]) -> DiagMatrix:
```

Comment:
Does torch sparse also support this?

Reply:
Also is this also supported for sparse matrices?

Reply:
Figured out that this is not supported by PyTorch sparse.  I'll close this PR for now.